Cli/Components/States/Results.razor

```diff
@@ -41,54 +69,127 @@
     protected override void OnInitialized()
     {
         ParseOptsToResultTypes();
-        GetAnalysisResults();
+        UpdatePageCount();
+        GetResultsPage(1);
         base.OnInitialized();
     }
 
-    void MoreAnalysisResults()
+    Dictionary<RESULT_TYPE, int> foundResultTypes = new Dictionary<RESULT_TYPE, int>();
+    public List<CompareResult> analysisResults = new List<CompareResult>();
+    int offset = 0;
+    int currentPage = 1;
+    int pageCount = 0;
+    List<string> pageStates = new List<string>();
+
+    int _maxResults = 100;
+    int MaxResults
     {
-        offset += count;
-        GetAnalysisResults();
+        get { return _maxResults; }
+        set
+        {
+            _maxResults = value;
+            UpdatePageCount();
+            GetResultsPage(1);
+        }
     }
 
-    int offset = 0;
-    int count = 100;
-
+    string _selectedResultType = "FILEMONITOR";
     string SelectedResultType
     {
         get { return _selectedResultType; }
-        set { _selectedResultType = value; offset = 0; GetAnalysisResults(); }
+        set
+        {
+            _selectedResultType = value;
+            UpdatePageCount();
+            GetResultsPage(1);
+        }
     }
 
-    string _selectedResultType = "FILEMONITOR";
+    void GetResultsPage(int selectedPage)
+    {
+        if (pageCount < 1) { return; }
+        SetPageState(selectedPage);
+        offset = MaxResults * (currentPage - 1);
+        GetAnalysisResults();
+        this.StateHasChanged();
+    }
+
+    void GetResultsPageByOffset(int pageOffset)
+    {
+        var selectedPage = currentPage + pageOffset;
+        if (selectedPage < 1) { selectedPage = 1;}
+        else if (selectedPage > pageCount) { selectedPage = pageCount; }
+        GetResultsPage(selectedPage);
+    }
+
+    string GetPageState(int selectedPage)
+    {
+        return pageStates[selectedPage - 1];
+    }
+
+    void SetPageState(int selectedPage)
```

Comment:
You can probably get rid of this function and just set the current page variable directly for clarity.

Reply:
Updated